File: AHYEON/10.GREEDY/pro_체육복.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    let n: Int = n
    let lost: [Int] = lost
    let reserve: [Int] = reserve
    var student: [Int] = [Int](repeating: 1, count: n)
    // 0: 없음 2: 여유있음

    for l in lost {
        student[l-1] = student[l-1] - 1
    }

    for r in reserve {
        student[r-1] = student[r-1] + 1
    }

    for (index, value) in student.enumerated(){
        if value == 0 {
            if (index > 0 && student[index - 1] == 2){
                student[index - 1] -= 1
                student[index] += 1
            }else if (index < student.count - 1) && student[index + 1] == 2{
                student[index + 1] -= 1
                student[index] += 1
            }
        }
    }

    var result = student.filter{($0 >= 1)}.count

    return result
}

File: 김순석/프로그래머스/pro_체육복.py
def solution(n, lost, reserve):
    answer = 0
    reserve.sort()
    lost.sort()
    new_reserve = []
    new_lost = []
    for i in range(len(reserve)):
        for j in range(len(lost)):
            if reserve[i] == lost[j]:
                lost[j] = 0
                reserve[i] = 0
                break
    for i in reserve:
        if i != 0:
            new_reserve.append(i)
    for i in lost:
        if i != 0:
            new_lost.append(i)

    for i in range(len(new_reserve)):
        for j in range(len(new_lost)):
            if new_reserve[i] - 1 == new_lost[j] and new_lost[j] != 0:
                new_lost[j] = -1
                break
            elif new_reserve[i] + 1 == new_lost[j] and new_lost[j] != 0:
                new_lost[j] = -1
                break

    cnt = len(new_lost) - new_lost.count(-1)
    answer = n - cnt
    return answer